snuba/datasets/discover.py

```diff
@@ -55,6 +61,67 @@
 logger = logging.getLogger(__name__)
 
 
+EVENT_CONDITION = FunctionCallMatch(
+    None,
+    Param("function", Or([StringMatch(op) for op in BINARY_OPERATORS])),
+    (
+        Or(
+            [
+                ColumnMatch(None, None, StringMatch("type")),
+                LiteralMatch(StringMatch("type"), None),
+            ]
+        ),
+        Param("event_type", Or([ColumnMatch(), LiteralMatch()])),
+    ),
+)
+
+
+def match_query_to_table(
+    query: Query, events_only_columns: ColumnSet, transactions_only_columns: ColumnSet
+) -> str:
+    has_event_columns = False
+    has_transaction_columns = False
+    for col in query.get_all_ast_referenced_columns():
+        if events_only_columns.get(col.column_name):
+            has_event_columns = True
+        elif transactions_only_columns.get(col.column_name):
+            has_transaction_columns = True
+
+    # Unless we have an impossible query, we only need to check which columns are referenced.
+    # If all columns are common, use the events table by default.
+    if not (has_event_columns and has_transaction_columns):
+        return TRANSACTIONS if has_transaction_columns else EVENTS
+
+    # In the case of an impossible query, check to see if the user has specified which type of event
+    # they want to see, and use that to inform the table selection. If there is no specification,
+    # default to events.
+    condition = query.get_condition_from_ast()
+    event_types = []
+    if condition:
+        for cond in condition:
+            result = EVENT_CONDITION.match(cond)
+            if not result:
+                continue
+
+            event_type_param = result.expression("event_type")
+            event_type = None
+            if isinstance(event_type_param, Column):
+                event_type = event_type_param.column_name
```

Comment:
what kind of query is this one ?
I am not sure I get when you can have a condition like 
`type=COLUMN_NAME`
Wouldn't this look if the column name is `transaction` ?

Reply:
I'm not sure, but there are some tests where the rhs of the condition is a Literal, and some where the rhs is a Column. I think it's because 'transaction' is a valid column. I didn't get a chance to dig into it.